lib/resty/iprepd.lua

```diff
@@ -191,17 +199,39 @@ end
 
 function _M.audit_log(self, ip)
   if self.audit_blocked_requests == 1 then
-    ngx.req.read_body()
+    local should_audit_request = 0
     local headers = ngx.req.get_headers()
-    local request_headers_all = ""
-    for k,v in pairs(headers) do
-      request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+    for i, uri in ipairs(self.audit_uri_list) do
+      if string.match(ngx.var.uri, uri) then
+        should_audit_request = 1
+        break
+      end
     end
-    local data = ngx.req.get_body_data()
-    if not data then
-      data = ngx.req.get_body_file()
+    if should_audit_request == 1 then
+      ngx.req.read_body()
+      local request_headers_all = ""
+      if self.audit_include_headers == 1 then
+        local headers = ngx.req.get_headers()
+        for k,v in pairs(headers) do
+          if k == "authorization" or k == "proxy-authorization" then
+            v = "REMOVED"
+          end
+          request_headers_all = request_headers_all .. string.format('"%s": "%s", ', k, v)
+        end
+      end
+      request_headers_all = ngx.encode_base64(request_headers_all)
+      local data = ngx.req.get_body_data()
+      if data then
+        data = ngx.encode_base64(data)
+      end
+      if not data then
+        data = ngx.req.get_body_file()
+      end
+      if not data then
```

Comment:
One other question here, so it looks like this will be applied to every request method. Should we also include the request method (e.g., GET, etc) in the log so it's clear why no body will be present for certain requests?

Also, is there a means to get the content length prior to requesting the actual request body? It might be worth wrapping the calls to `get_body_data` and `get_body_file` in a conditional around there being content present.

Reply:
Cleaned up the log format. The request with method is part of the log line already.

There isn't an api call that allows us to get content length. Could check headers or only do the calls for certain request methods...

Reply:
Sounds good, will leave final r+ up to @ajvb